controllers/networkfence_controller.go

```diff
@@ -193,3 +193,40 @@ func (r *NetworkFenceReconciler) sendUnfenceNetworkReq(ctx context.Context, nwFe
 
 	return nil
 }
+
+// addFinalizerToNetworkFence adds a finalizer to the Networkfence instance.
+func (r *NetworkFenceReconciler) addFinalizerToNetworkFence(
+	ctx context.Context,
+	logger *logr.Logger,
+	networkFence *csiaddonsv1alpha1.NetworkFence) error {
+
+	if !util.ContainsInSlice(networkFence.Finalizers, networkFenceFinalizer) {
+		logger.Info("adding finalizer to NetworkFence object", "Finalizer", networkFenceFinalizer)
+
+		networkFence.Finalizers = append(networkFence.Finalizers, networkFenceFinalizer)
+		if err := r.Client.Update(ctx, networkFence); err != nil {
```

Comment:
use `Patch` instead of `Update`?

Reply:
As per the discussions on CSI Addons controller https://github.com/csi-addons/kubernetes-csi-addons/pull/58#discussion_r781303210 and to maintain the same coding style, I stuck with Update